### DISCUSSION TITLE
Removing  *-blx.bib files

### DISCUSSION THREAD

**Discussion by anton-vrba** [WinEdt-Team/WinEdt-Support#46] [1.1 General Help and Discussion]
Hi Alex,

When using the package `biblatex`  this package generates a file with the extension `-blx.bib` .  This file is not deleted with the [Erase Output Files]  button (the dustbin button). 

Is there a simple way to include this file in the list of files to be deleted?

When making changes to the master bib file, these changes, for some strange reason or other, only migrate into the document when both the -blx.bib and .bbl files are deleted.

regards
Anton

**Comment by Alex-Simonic** [comment]
In the default settings, this file is included. Options Interface -> Advanced -> Common Macros -> Erase Working Files.
In line 38 you should find:

```
    AddFileItem(1,"RUN Files","%N.run.xml","TeX");
    AddFileItem(1,"BLX Files","%N-blx.bib","TeX");
```
If you have a customized version of this file you can add it to your list of files that should be deleted.

Here is a snapshot of the default 11.2 settings in action:

![image](https://github.com/WinEdt-Team/WinEdt-Support/assets/157834149/56c743b2-4b8b-4979-aafe-fa3138e53b13)

**Reply by Alex-Simonic** [reply]
Furthermore, if you make changes to .bib sources and are using [PDF]TeXify or Compile, you should execute BibTeX (Ctrl+Shift+B) before recompiling. This will ensure that changes are incorporated even if you don't erase working files. Disabling the "Quick TeXify" option in the Execution Modes| TeX Options might have the same effect. Still, then BibTeX/ Biber will be executed before every compilation (even if it is not required). With BibTeX this is not a big problem but Biber is much slower.

**Comment by anton-vrba** [comment]
Thank you Alex, did not know that  much easier than deleting all the 
output files.

**Comment by anton-vrba** [comment]
Thanks Alex, I found the two lines

     AddFileItem(1,"RUN Files","%N.run.xml","TeX");
     AddFileItem(1,"BLX Files","%N-blx.bib","TeX");

the xml file deletes OK but the blx.bib does not deleted on my system.
So I changed the BLX line to

     AddFileItem(1,"BLX Files","*-blx.bib","TeX");

and the file deletes OK

regards
Anton

**Comment by anton-vrba** [comment]
I now found the reason why the -blx.bib file did not delete.  My file 
name contains a blank which are replaced by underscores in the blx.bib 
file.  Hence %N produced a file name "file name-blx.bib" with blank 
whereas the file to delete is "file_name-blx.bib" with underscore.

**Reply by Alex-Simonic** [reply]
It is always good to know the reason for such oddities. I was puzzled by this as it works fine with default settings but now we know why (I did not test with spaces in the filename). Putting * instead of %N is the best solution to address this biblatex oddity (with [backend=bibtex])... Now we know that it is not a WinEdt issue.
